SteamReview/SteamReview/Steam.cpp

```diff
@@ -226,19 +246,31 @@ bool FSteam::AddNewGameToCategory(const FGame InGame)
 
 			int CategoryChosen;
 			std::cin >> CategoryChosen;
-			CategoryChosen = ValidateInput(CategoryChosen, 1, CategoryContainer.GetCurrentCategoryAmount()) - 1;
+			ValidateInput(CategoryChosen, 1, CategoryContainer.GetCurrentCategoryAmount());
+			
+			//Subtract 1 to select the current object inside of the array
+			CategoryChosen = CategoryChosen - 1;
 
 			//adding the game to category
-			if (CategoryContainer.AddGameToCategory(CategoryChosen, InGame))
+			FCategory Category;
+			if(CategoryContainer.GetCategory(CategoryChosen, Category))
 			{
-				std::cout << "------------------------------------------\n" << std::endl;
-				std::cout << "The game is going to add to the category: " << CategoryContainer.GetCategory(CategoryChosen - 1).GetCategoryName() << std::endl;
-				return true;
+				if (CategoryContainer.AddGameToCategory(CategoryChosen, InGame))
+				{
+					std::cout << "------------------------------------------\n" << std::endl;
+					std::cout << "The game is going to add to the category: " << Category.GetCategoryName() << std::endl;
+					return true;
+				}
+				else
+				{
+					std::cout << "The game couldn't be added to the category: " << Category.GetCategoryName() << std::endl;
+					return false;
+				}
 			}
 			else
 			{
 				std::cout << "------------------------------------------\n" << std::endl;
-				std::cout << "The game couldn't be added to the category: " << CategoryContainer.GetCategory(CategoryChosen - 1).GetCategoryName() << std::endl;
+				std::cout << "Something is wrong" << std::endl;
```

Comment:
Something WENT wrong, you might want to add a lil more info into this, if it is already being shown to the user, perhaps be clearer on what it is exactly what went wrong in here, so he could know how to ask for assistance.